Add Typography

Jolg42 opened the pull request:
**[https://github.com/Jolg42/awesome-typography](https://github.com/Jolg42/awesome-typography)**

**I created this list in June 2016, the list now contains +300 resources that are useful when programming with fonts. This repository received multiple PR by third parties and more and more people are using it now!**

_I just reorganized & cleaned the list following guidelines below._ 😇

# By submitting this pull request I confirm I've read and complied with the below requirements.

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title.<br>For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with these requirements:
	- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
	- It's the result of hard work and the best I could possibly produce.
	- Non-generated Markdown file in a GitHub repo.
	- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
	- Not a duplicate.
	- Only has awesome items. Awesome lists are curations of the best, not everything.
	- Includes a project logo/illustration whenever possible.
		- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
		- The image should link to the project website or any relevant website.
		- The image should be high-DPI. Set it to maximum half the width of the original image.
	- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
	- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
		- Should be placed on the right side of the readme heading.
		- Should link back to this list.
	- Has a Table of Contents section.
		- Should be named `Contents`, not `Table of Contents`.
		- Should be the first section in the list.
		- Should only have one level of sub-lists, preferably none.
	- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
		- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
		- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
		- If you use a license badge, it should be SVG, not PNG.
	- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
		- The file should be named `contributing.md`. Casing is up to you.
	- Has consistent formatting and proper spelling/grammar.
		- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
		- The description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
	- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Go to the top and read it again.

Jolg42 commented:
Hi @sindresorhus!

I guess you're busy...! I saw you added some comments to more recent PR and I was thinking that maybe you missed mine.

Let me know if there is something I can do 😉

legodude17 commented:
This doesn't make sense: `Hasklig - Hasklig - a code font with monospaced ligatures.` Maybe just `Hasklig - Code font with monospaced ligatures.`

Might want a description: `Firefox OS Emojis.`  What is Firefox OS, and why are it's emojis mixed with a bunch of fonts?

Maybe change the name to `awesome-programming-typography`? Just a suggestion though.

Add more descriptions please.

legodude17 commented:
What do you mean by that? @gdonnini

gdonnini commented:
To avoid email spam to everyone subscribed to this list, we should
implement standards around appropriate edits / responses / reply-alls

On Thu, May 3, 2018 at 2:31 PM, Joshua Bennett <notifications@github.com>
wrote:

> What do you mean by that? @gdonnini <https://github.com/gdonnini>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sindresorhus/awesome/pull/1247#issuecomment-386442826>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADEXEmsZ70qNwkJBNiEUv5Ua-c-QqYmAks5tu3chgaJpZM4StUTQ>
> .
>



-- 
Gabriel Donnini
E-mail: gabedonnini@gmail.com
Cell: 617-331-7063

Jolg42 commented:
Hi @legodude17 😉 

Thanks for the suggestion about `Hasklig` I'll fix that asap.
About Firefox OS, it was an OS that is now deprecated but somebody forked it more recently I think... So this link is the repository for the emojis designed for that OS. And the emojis are actually part of a font!

Jolg42 commented:
I just fixed the `Hasklig` typo and added a small description for the `Firefox OS Emojis` entry.
@legodude17 @gdonnini Do you think this list can be added now? 😇

RoelN commented:
+1

zachleat commented:
Would love to see this added!

sindresorhus commented:
Hey, sorry for the lack of response, please see #1363.

nschloe commented:
Hi! I've reviewed this submission and it has a few issues.

:white_check_mark: Good!
:x: Needs work!
:grey_question: Not sure, not required, or mixed results

Reviewer comments are in bold after an item.

#### Awesome Requirements
:white_check_mark: Has been around for at least 30 days.
:white_check_mark: It's the result of hard work and the best I could possibly produce.
:white_check_mark: Non-generated Markdown file in a GitHub repo.
:grey_question: Includes a succinct description of the project/theme at the top of the readme. 
**The list deals with _digital_ typography; that keyword should probably appear in the description.**
:white_check_mark: The repo should have awesome-list & awesome as GitHub topics.
:x: Not a duplicate. **Check https://github.com/deanhume/typography#readme.**
:white_check_mark: Only has awesome items. Awesome lists are curations of the best, not everything.
:white_check_mark: Includes a project logo/illustration whenever possible. Not required
:white_check_mark: Entries have a description, unless the title is descriptive enough by itself.
:white_check_mark: Includes the [Awesome badge]
:white_check_mark: Should be placed on the right side of the readme heading.
:white_check_mark: Should link back to this list.
:white_check_mark: Has a Table of Contents section.
:white_check_mark: Should be named Contents, not Table of Contents.
:white_check_mark: Should be the first section in the list.
:grey_question: Should only have one level of sub-lists, preferably none. "Tools with GUI" has a sublist, I guess that's okay.
:grey_question: Has an appropriate license. **In addition to the linked "Public domain" image, it should be explicitly stated under which license this list is. For example, the original awesome list has**
> To the extent possible under law, Sindre Sorhus has waived all copyright and related or neighboring rights to this work.

:white_check_mark: Has contribution guidelines.
:white_check_mark: The file should be named contributing.md. Casing is up to you.
:white_check_mark: Has consistent formatting and proper spelling/grammar. :white_check_mark: The link and description are separated by a dash.
:white_check_mark: The description starts with an uppercase character and ends with a period.
:white_check_mark: Consistent and correct naming. For example, Node.js, not NodeJS or node.js.
:white_check_mark: Doesn't include a Travis badge.

#### Additional Comments

The existence of a web typography list is a bit of a bummer, and there's indeed an overlap (e.g., Typeset.js). Not sure what to do about this.

In addition,
```
awesome-lint README.md
```
gives many errors, mostly with trivial fixes.

Jolg42 commented:
Thanks a lot for the review @nschloe!
I'll check your report and make the appropriate changes 👍

About the existence of https://github.com/deanhume/typography#readme, the list has a clear focus, see screenshot:
<img width="684" alt="screenshot 2018-07-29 at 14 49 23" src="https://user-images.githubusercontent.com/1328733/43366457-ad776f8c-933e-11e8-938c-2d16a2922115.png">

My list is more about tech resources, libraries, tools for digital typography, so both could be listed I think as they are really different.

You also mentioned that there is an overlap but it's only a few links (Maybe 1 to 5?) over +300 awesome resources that are not overlapping.

What do you think?

nschloe commented:
> What do you think?

Just to be clear, I like this list here better than the web-typography one. The best solution would perhaps be a web typography subsection your list. Not sure if this is possible. Admitting overlap is a bit of a slippery slope. Soon enough, somebody will create a `retina-typography` or a `mobile-typography` one, all with some overlap.

Jolg42 commented:
So if I add a Web-Typography (and a link to deanhume/typography#readme?) section that would mean deleting the other list?

I'm ok with that but I just need a confirmation before updating it :)

nschloe commented:
Eventually, this is @sindresorhus's call to make.

sindresorhus commented:
Can you fix the merge conflict?

sindresorhus commented:
> * **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)

sindresorhus commented:
```
❯ awesome-lint https://github.com/Jolg42/awesome-typography
✖ Linting

  README.md:228:124
  ✖   47:120  List item description contains invalid markdown                    remark-lint:awesome/list-item
  ✖   80:54   List item description must start with valid casing                 remark-lint:awesome/list-item
  ✖   93:52   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  108:65   List item description must end with proper punctuation             remark-lint:awesome/list-item
  ✖  118:59   Text "github" should be written as "GitHub"                        remark-lint:awesome/spell-check
  ✖  167:57   List item description must start with valid casing                 remark-lint:awesome/list-item
  ✖  173:59   Text "OSX" should be written as "macOS"                            remark-lint:awesome/spell-check
  ✖  222:39   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  228:124  Link to http://pikensoft.com/programs/OpenTypeTemplate.bt is dead  remark-lint:no-dead-urls
  ✖  233:54   Text "Mac OS X" should be written as "macOS"                       remark-lint:awesome/spell-check
  ✖  240:41   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  242:84   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  272:118  List item description must end with proper punctuation             remark-lint:awesome/list-item
  ✖  273:80   List item description must end with proper punctuation             remark-lint:awesome/list-item
  ✖  274:48   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  301:88   Text "Mac OS X" should be written as "macOS"                       remark-lint:awesome/spell-check
  ✖  315:49   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  316:52   List item link and description must be separated with a dash       remark-lint:awesome/list-item
  ✖  322:56   Text "OS X" should be written as "macOS"                           remark-lint:awesome/spell-check
  ✖  362:2    License image must be SVG                                          remark-lint:awesome/license

  20 errors
```

https://github.com/sindresorhus/awesome-lint

sindresorhus commented on readme.md:
```diff
@@ -407,6 +407,7 @@
 - [Audio Visualization](https://github.com/willianjusten/awesome-audio-visualization)
 - [Broadcasting](https://github.com/ebu/awesome-broadcasting)
 - [Pixel Art](https://github.com/Siilwyn/awesome-pixel-art) - Pixel-level digital art.
+- [Typography](https://github.com/Jolg42/awesome-typography) - TrueType & OpenType.
```
Needs a more elaborate description.

sindresorhus commented:
Closing for lack of response.
